torch/_inductor/codegen/triton.py

```diff
@@ -2367,9 +2367,17 @@ def _lift_helper(self, fn, num_args) -> str:
         cse = CSE(prefix="", suffix="")
         overrides = TritonOverrides(V.MockHandler())
 
+        # Build a name that changes depending on fn to workaround a triton bug
+        # where the combine_fn to reduce and scan is not hashed, and so different
+        # scan ops may collide in the triton cache.
+        # This is fixed with the latest triton pin, but not the triton-rocm pin.
+        helper_name = "_triton_helper_fn"
+
```

Comment:
Rather than making the code more complex, we can probably just remove these workarounds and skip rocm really.

Reply:
It's a pretty small change and also with backward compatibility for older triton verisons which is required because meta internal isn't kept in sync with oss.